cranelift/fuzzgen/src/function_generator.rs

```diff
@@ -10,40 +11,32 @@ use cranelift::frontend::{FunctionBuilder, FunctionBuilderContext, Variable};
 use cranelift::prelude::{
     EntityRef, InstBuilder, IntCC, JumpTableData, StackSlotData, StackSlotKind,
 };
+use cranelift_entity::ListPool;
 use std::ops::RangeInclusive;
 
 type BlockSignature = Vec<Type>;
 
-fn insert_opcode_arity_0(
-    _fgen: &mut FunctionGenerator,
-    builder: &mut FunctionBuilder,
-    opcode: Opcode,
-    _args: &'static [Type],
-    _rets: &'static [Type],
-) -> Result<()> {
-    builder.ins().NullAry(opcode, INVALID);
-    Ok(())
-}
-
-fn insert_opcode_arity_2(
+fn insert_opcode(
     fgen: &mut FunctionGenerator,
     builder: &mut FunctionBuilder,
     opcode: Opcode,
     args: &'static [Type],
     rets: &'static [Type],
 ) -> Result<()> {
-    let arg0 = fgen.get_variable_of_type(args[0])?;
-    let arg0 = builder.use_var(arg0);
-
-    let arg1 = fgen.get_variable_of_type(args[1])?;
-    let arg1 = builder.use_var(arg1);
+    let mut pool = ListPool::new();
```

Comment:
Hmm, I'm surprised that this works, and I suspect it does only accidentally: we should be building the argument list in the pool owned by the `DataFlowGraph`. The `ValueList` is just an index into the pool, so `ValueList`s need to be built with the proper pool and will silently fail if referring to another pool. (This is a kind of unfortunate logical-unsafety but the type system doesn't give us a way to tie the list to a *particular* pool.)

Reply:
Fixed. I wonder how the fuzzer didn't immediately catch this, I only ran it for a couple of minutes after making this change, but I would have expected it to fail instantly.

Thanks!